lib/nerves_ssh/options.ex

```diff
@@ -289,4 +294,21 @@ defmodule NervesSSH.Options do
   else
     defp decode_key(key), do: :public_key.ssh_decode(key, :auth_keys)
   end
+
+  defp generate_system_keys!() do
+    File.mkdir_p!(@default_device_system_dir)
+    key = X509.PrivateKey.new_ec(:ed25519)
+    pubkey = X509.PublicKey.derive(key)
+
+    _ =
+      File.write!(
+        "#{@default_device_system_dir}/ssh_host_ed25519_key",
+        X509.PrivateKey.to_pem(key)
+      )
+
+    File.write!(
+      "#{@default_device_system_dir}/ssh_host_ed25519_key.pub",
+      X509.PublicKey.to_pem(pubkey)
+    )
```

Comment:
I think this may not be used. In fact, it would make sense for the public key not to be needed since `:ssh` should only need to read the private key to work.

The reason that I think this is that ssh public keys aren't stored as `.pem` files, so any code that reads them should break. The `:ssh_file` module has a utility function for saving public keys in the ssh format, but it doesn't work with ed25519. Here's an RSA example:

```elixir
iex> key = X509.PrivateKey.new_rsa(512)
{:RSAPrivateKey, :"two-prime",
 9769417810301217260779373416366533577363742994075998014340835447000850344245664864644909295432854236562818424745064045565488616792581696690756590872251901,
 65537,
 7508515420371275972739933762338561366736526460039489448439017371338828933879246466175970192481706203889848112177688200586190858898287217124536133126684793,
 102448277959017751041140026282525729773655474408125414564998593478258273482867,
 95359512184375266216918566931457327993356106224618445377243090513898278269903,
 47953116173624495032394089540912145605332183849484340406120158895571216219241,
 68393098718500618173856604971942875703796487739474882989943607832449985074057,
 62761306075489818964602436552462199901787047072208554755977440215961420010016,
 :asn1_NOVALUE}
iex> pubkey = X509.PublicKey.derive(key)
{:RSAPublicKey,
 9769417810301217260779373416366533577363742994075998014340835447000850344245664864644909295432854236562818424745064045565488616792581696690756590872251901,
 65537}
iex> :ssh_file.encode([{pubkey, []}], :openssh_key)
"ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAAAQQC6h+tnpUOqSe1tfXz0MVnjuYFhRDMXBOsAlW1IJNDtksZiSRx0OxCWQ2LuFK/FbWKOJgnpFTYjeYZ0SRtaM039 \n"

Reply:
So I tested this and it looks like if works without public keys if it is a format that `:ssh_file.encode` supports. So RSA works by default, but `ed25519` breaks.

However, if the pub key is supplied, then everything works

Reply:
A lot of research went in around this. Some OTP PR's are coming and documentation in the next PR about options with ed25519